nexus/pkg/artifacts/artifacts.go

```diff
@@ -250,3 +257,52 @@ func (as *ArtifactSaver) Save() (ArtifactSaverResult, error) {
 
 	return ArtifactSaverResult{ArtifactId: artifactId}, nil
 }
+
+func (al *ArtifactLinker) Link() error {
+	client_id := al.Artifact.ClientId
+	server_id := al.Artifact.ServerId
+	portfolio_name := al.Artifact.PortfolioName
+	portfolio_entity := al.Artifact.PortfolioEntity
+	portfolio_project := al.Artifact.PortfolioProject
+	portfolio_aliases := []gql.ArtifactAliasInput{}
+
+	for _, alias := range al.Artifact.PortfolioAliases {
+		portfolio_aliases = append(portfolio_aliases,
+			gql.ArtifactAliasInput{
+				ArtifactCollectionName: portfolio_name,
+				Alias:                  alias,
+			},
+		)
+	}
+	var err error
+	var response *gql.LinkArtifactResponse
+	if server_id != "" {
+		response, err = gql.LinkArtifact(
+			al.Ctx,
+			al.GraphqlClient,
+			portfolio_name,
+			portfolio_entity,
+			portfolio_project,
+			portfolio_aliases,
+			nil,
+			&server_id,
+		)
+	} else if client_id != "" {
```

Comment:
I don't know that we have an upstream check to ensure that client and server id weren't both omitted; I guess that would be okay and we just have a no-op? It seems like we should return an error in that case though.

Reply:
We have a gorilla check that makes sure that at both client_id and artifact_id weren't null here: https://github.com/wandb/core/blob/790eee803599fec84dd171da3f65a7aa7a0e7b1c/services/gorilla/api/resolver/mutation_artifact_collections.go#L503-L505